frappe/model/meta.py

```diff
@@ -452,14 +448,56 @@ def init_field_caches(self):
 			self._table_fields = self.get("fields", {"fieldtype": ["in", table_fields]})
 
 	def sort_fields(self):
-		"""Sort custom fields on the basis of insert_after"""
+		"""
+		Sort fields on the basis of following rules (priority descending):
+		- `field_order` property setter
+		- `insert_after` computed based on default order for standard fields
+		- `insert_after` property for custom fields
+		"""
+
+		if field_order := getattr(self, "field_order", []):
+			field_order = [fieldname for fieldname in json.loads(field_order) if fieldname in self._fields]
+
+			# all fields match, best case scenario
+			if len(field_order) == len(self.fields):
+				self._update_fields_based_on_order(field_order)
```

Comment:
This is incorrect. Same length != same content? 

Compare `set` of fieldnames

Reply:
Edge case: 1 field is added and 1 field is removed.

Reply:
`self._fields` is just a dict version of `self.fields`. In line 459 (just above this), I am filtering `field_order` based on `self._fields`. So it cannot contain anything other than the contents of `self.fields` / `self._fields`. So length match should be adequate.